### DISCUSSION TITLE
Essencial features

### DISCUSSION THREAD

**Discussion by Artars** [gilzoide/raise-and-shine#2] [Ideas]
Some features that may be useful

1) 2D Texture visualization
- Switch between albedo, heightmap and normal map
2) 3D Visualization
- Visualize plane
- Toggle heightmap, normal and albedo
3) 2D sprite illumination
- Visualize sprite with 2D illumination
- Change light position
- Toggle normal map
4) Heightmap painting
- Paint / Erase pencil with pen size, force and maybe falloff
- Selection tool, to mask operations
- Gradient tool / Bevel : control minimum and maximum, transition type (linear, radial, following mask)
- Profile editor:
  Select two points in the texture, you will be able to see and edit the "profile" of height between then. Imagine as a terrain profile.
5) Export heightmap texture and derived normal map
6) Import albedo and heightmap

**Comment by gilzoide** [comment]
Good points!
I'm thinking if it would be nice to be able to edit normals directly instead of hard linking them to the height map. I guess this link would be the most used workflow and should be the default, but maybe fine tunning normals directly would be useful in some cases.
